src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -101,18 +102,47 @@ const DataProcessingPage = () => {
       <Col span='8'>
         <Select
           value={stepId}
-          onChange={(id) => setStepId(id)}
+          onChange={(id) => {
```

Comment:
if you are renaming `id` to `key` in the `steps` array above, it makes sense to rename these to `key` too

Reply:
so this is actually not the `key`, but the step number, where the number represents the ordering of execution for a step. We definitely need to rename this to make it more obvious